src/ui/public/chrome/api/__tests__/nav.js

```diff
@@ -45,6 +45,36 @@ describe('chrome nav apis', function () {
     });
   });
 
+  describe('#getNavLinkById', () => {
+    it ('retrieves the correct nav link, given its ID', () => {
+      const appUrlStore = new StubBrowserStorage();
+      const nav = [
+        { id: 'kibana:discover', title: 'Discover' }
+      ];
+      const { chrome, internals } = init({ appUrlStore, nav });
+
+      const navLink = chrome.getNavLinkById('kibana:discover');
+      expect(navLink).to.not.be(undefined);
+      expect(navLink.title).to.be('Discover');
```

Comment:
I _think_ these two expectations can be replaced with:

```javascript
expect(navLink).to.eql(nav[0]);
```

This should do a deep comparison: http://chaijs.com/api/bdd/#method_eql